mono/metadata/cominterop.c

```diff
@@ -2742,6 +2742,42 @@ init_com_provider_ms (void)
 }
 
 gpointer
+mono_buffer_to_bstr (MonoArray* ptr, int slen)
```

Comment:
This code is C&P from mono_string_to_bstr, we should refactor mono_string_to_bstr to take pointer + length arguments and use that instead.